src/nodes.ts

```diff
@@ -15,6 +17,20 @@ export abstract class Node {
     this.range = [start, end];
     this.raw = raw;
   }
+
+  getChildren(): Array<Node> {
+    let children: Array<Node> = [];
+    for (let childField of this.getChildFields()) {
+      if (Array.isArray(childField)) {
+        children.push(...childField.filter<Node>((node): node is Node => node !== null));
```

Comment:
Yep, it was necessary. I think it used to be that `filter` couldn't be used for type refinement at all, but now the current state of things is that if the callback is a type guard, then it'll work. But it won't infer your callback to be a type guard, so you need to specify the return type explicitly.

(Also, minor, but `filter(node => node)` also doesn't typecheck because the predicate needs to return a boolean.)

This issue has some discussion about it: https://github.com/Microsoft/TypeScript/issues/7657